connectors/src/connectors/slack/bot.ts

```diff
@@ -710,5 +712,13 @@ async function makeContentFragment(
     content: text,
     url: url,
     contentType: "slack_thread_content",
+    context: {
+      timezone: slackChatBotMessage.slackTimezone || "Europe/Paris",
+      username: slackChatBotMessage.slackUserName,
+      fullName:
+        slackChatBotMessage.slackFullName || slackChatBotMessage.slackUserName,
+      email: slackChatBotMessage.slackEmail,
+      profilePictureUrl: slackChatBotMessage.slackAvatar || null,
+    },
   } as PostContentFragmentRequestBody);
```

Comment:
Why the "as" here ? can't we have type-checking ?

Reply:
@fontanierh I don't see why this removes type checking.
This is similar to typing the function explicitly. If something is not compliant with the type `PostContentFragmentRequestBody`, it will complain.

Reply:
No it is not. `as X` means "STFU typescript, this is what I say it is, don't check it"

Reply:
I suppose it depends on the situation, but if typescript can do type checking it will do it apparently.
If I have a typo in my payload, it will complain.
But I understand this might not be the most expressive way to do that, and might lead to error.


<img width="1739" alt="Screen Shot 2023-11-16 at 11 14 51" src="https://github.com/dust-tt/dust/assets/358965/5c0fc3c3-c484-473c-b3e6-65a6ac46e8e3">

Reply:
Fixed.

Reply:
Interesting, thanks ! Didn't know that it "sometimes" checks 🤔  Not sure I fully understand how this works

Reply:
Same. I would like to understand what are the rules here. But I also agree that this can be miss leading so I can stop doing that.

Reply:
I've definitely seen a bunch of cases where it allows to lie about the type. 
For instance, this:
<img width="419" alt="Screenshot 2023-11-16 at 11 49 23" src="https://github.com/dust-tt/dust/assets/14199823/089874ea-ff29-4b79-9bf4-2238e7f82ebd">